client/src/components/Allocations/AllocationsModals/AllocationsManageTeamModal/AllocationsManageTeamModal.jsx

```diff
@@ -0,0 +1,194 @@
+              {` ${user.username} (${user.email})`}
+            </span>
+          );
+        },
+      },
+      {
+        Header: 'Role',
+        accessor: ({ role }) => {
+          switch (role) {
+            case 'PI':
+              return 'Principal Investigator';
+            case 'Delegate':
+              return 'Allocation Manager';
+            default:
+              return 'Member';
+          }
+        },
+      },
+      {
+        Header: '',
+        accessor: 'id',
+        Cell: (el) => {
+          const deleteOperationOccuring =
+            removingUserOperation.loading &&
+            el.row.original.username === removingUserOperation.userName;
+          const deleteOperationFailed =
+            removingUserOperation.error &&
+            el.row.original.username === removingUserOperation.userName;
+          const removable =
+            !deleteOperationOccuring && el.row.original.role !== 'PI';
+
+          return (
+            <>
+              {deleteOperationFailed && (
+                <Message type="error">Something went wrong.</Message>
+              )}
+              {deleteOperationOccuring && <LoadingSpinner placement="inline" />}
+              {removable && (
+                <Button
+                  color="link"
+                  disabled={removingUserOperation.loading}
+                  onClick={(e) => {
+                    dispatch({
+                      type: 'REMOVE_USER_FROM_TAS_PROJECT',
+                      payload: {
+                        projectId,
+                        id: el.row.original.username,
+                      },
+                    });
+                  }}
+                >
+                  Remove
+                </Button>
+              )}
+            </>
+          );
+        },
+      },
+    ],
+    [rawData, removingUserOperation]
+  );
+  const { getTableProps, getTableBodyProps, rows, prepareRow, headerGroups } =
+    useTable({
+      columns,
+      data,
+    });
+  return (
+    <Table hover responsive borderless size="sm" {...getTableProps()}>
+      <thead>
+        {headerGroups.map((headerGroup) => (
+          <tr {...headerGroup.getHeaderGroupProps()}>
+            {headerGroup.headers.map((column) => (
+              <th {...column.getHeaderProps()}>{column.render('Header')}</th>
+            ))}
+          </tr>
+        ))}
+      </thead>
+      <tbody {...getTableBodyProps()}>
+        {rows.map((row) => {
+          prepareRow(row);
+          return (
+            <tr>
+              {row.cells.map((cell) => (
+                <td {...cell.getCellProps()}>{cell.render('Cell')}</td>
+              ))}
+            </tr>
+          );
+        })}
+      </tbody>
+    </Table>
+  );
+};
+
+const AllocationsManageTeamModal = ({
+  isOpen,
+  toggle,
+  projectId,
+  viewToggle,
+}) => {
+  const dispatch = useDispatch();
+
+  const { teams, loadingUsernames, search } = useSelector(
```

Comment:
no errors gotten from state or used later to display an error message to users like in AllocationsTeamViewModel.  Specifically, if the listing of users fails.  Maybe the though process was that listing of users should be working cause the user got to this modal from the AllocationsTeamViewModal. but probably should be changed as if the listing of users occurs again after users are added and if that fails, then we get a spinner that does not go away